### ISSUE TITLE
eFolder Express is returning .bin files

### ISSUE THREAD

**Issue by shanear** [department-of-veterans-affairs/caseflow-efolder#404]
From Nick Holtz (i don't know his github tag). He's seeing that efolder express is downloading .bin files in some of the eFolders.

Get the file numbers where he is experiencing this issue, and figure out why this is happening, and if there's anything we can do to fix it.

# Notes
- Idea:  put monitors in the app.

**Comment by nicholasholtz** [comment]
I have file number for whoever would like it.

**Comment by nicholasholtz** [comment]
![image](https://cloud.githubusercontent.com/assets/16781398/23524080/30521d3a-ff57-11e6-8094-cd1470de25af.png)
@shanear

**Comment by aroltsch** [comment]
Some documents are returned with mime_type: application/octet-stream which in turn becomes a document with a `.bin` extension . In VBMS, these documents are displayed without error. I suggest that we add a check for `application/octet-stream` mime_type and use `.pdf` as a preferred extension instead of `.bin`. Thoughts? 
@amprokop @shanear

**Comment by amprokop** [comment]
@aroltsch — let's try to get an answer from VBMS first before we take more action. link up with @kierachell on this and add their response here?

**Comment by aroltsch** [comment]
Waiting on the response from the VBMS team.

**Comment by kierachell** [comment]
I sent them an email in the context of testing the new API:

```
- We were testing FindDocumentSeriesReference and found a few documents return with mime type "application/octet-stream" and file extension `.bin`. It seems like these documents can be opened with Adobe if the file extension is changed to `.pdf`. Is it expected that some of the documents in eFolder will be of mimetype "application/octet-stream" and safe to convert to PDF?
```

Meanwhile:

Here are some cases that are returning errors:
- 332345345   

`TypeError: no implicit conversion of Symbol into Integer` in xml_helper


- 112938488
`<env:Envelope xmlns:env="http://schemas.xmlsoap.org/soap/envelope/"><env:Header/><env:Body><env:Fault><faultcode>env:Server</faultcode><faultstring>Business Errors GUID: 93c7a5db-0bac-418b-808e-cbb1afa67797</faultstring><detail><ns4:EFolderFault xmlns:ns4="http://service.efolder.vbms.vba.va.gov/common"><ns4:code>EFSERR1001</ns4:code><ns4:message>The System has encountered an unknown error. Please contact your administrator.</ns4:message><ns4:type>System</ns4:type><ns4:uuid>93c7a5db-0bac-418b-808e-cbb1afa67797</ns4:uuid></ns4:EFolderFault></detail></env:Fault></env:Body></env:Envelope>`

**Comment by aroltsch** [comment]
Since @kierachell received no response, I am opening an issue in the dsva-vbms repo.

**Comment by aroltsch** [comment]
Response from the VBMS team:

```
We reached out to the Dev team for confirmation on our answer.  Yes it is expected that some of the documents may have a MIME type of "application/octet-stream".  However, there is not a guarantee that all documents of this type can be opened as PDFs.  The "application/octet-stream" MIME type could represent arbitrary data, mislabeled PDFs, etc. and there is no guarantee on file format without investigating the individual document.
```

**Comment by kierachell** [comment]
Currently it's causing 
https://sentry.ds.va.gov/department-of-veterans-affairs/efolder/issues/117/

Because the files do not exist locally or on S3

Check out download 19157. It has 2 `.bin` files that cause alerts

**Comment by shanear** [comment]
@aroltsch may want to revert and deploy, or if you think the fix will be quick we can get that in. Would be nice to have whatever option we choose before tomorrow morning's deploy

**Comment by amprokop** [comment]
Our hypothesis is that https://github.com/department-of-veterans-affairs/caseflow-efolder/pull/462, now merged, should fix the issue going forward.

**Comment by kierachell** [comment]
So, is there stuff being reverted? I provided 
`{F2D74AB4-7B7C-42A3-BDB8-5554E2778AAD} is of type application/octet-stream` from preprod

I can try redownloading some of the prod .bins to validate. Just let me know if my input is needed

**Comment by aroltsch** [comment]
@kierachell, we are not reverting this ticket. We were talking of reverting #462 and use `after_initialize` hook instead of `before_create`

**Comment by amprokop** [comment]
I haven't seen any errors related to this in Sentry over the last 36 hours. @aroltsch is there some reason we wanted to revert 462, or that we think it didn't work?

**Comment by shanear** [comment]
is not that #462 doesn't work. It's that it is no longer needed once all downloads are processed through the new .bin code, so it's overkill to keep it in there.

**Comment by aroltsch** [comment]
I honestly think `after_initialize ` is overkill because `after_initialize` callback is triggered for each object that is found and instantiated. I would think in our case we would only like to set mime type once and forget about it forever. 
@shanear @amprokop
